addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -0,0 +1,131 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedClusterAddOn is the Custom Resource object which holds the current state
+// of an add-on. This object is used by add-on operators to convey their state.
+// The resource should be created in the ManagedCluster's cluster namespace.
+type ManagedClusterAddOn struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// spec holds configuration that could apply to any operator.
+	// +kubebuilder:validation:Required
+	// +required
+	Spec ManagedClusterAddOnSpec `json:"spec"`
+
+	// status holds the information about the state of an operator.  It is consistent with status information across
+	// the Kubernetes ecosystem.
+	// +optional
+	Status ManagedClusterAddOnStatus `json:"status"`
+}
+
+// ManagedClusterAddOnSpec is empty for now, but you could imagine holding information like "pause".
+type ManagedClusterAddOnSpec struct {
+}
+
+// ManagedClusterAddOnStatus provides information about the status of the operator.
+// +k8s:deepcopy-gen=true
+type ManagedClusterAddOnStatus struct {
+	// conditions describe the state of the operator's managed and monitored components.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []AddOnStatusCondition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+
+	// addOnResource is a reference to the detailed resource driving the add-on
+	// this resource must be located in the same namespace as the ManagedClusterAddOn
+	// +required
+	AddOnResource ObjectReference `json:"addOnResource"`
+}
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
+	// group of the referent.
+	// +kubebuilder:validation:Required
+	// +required
+	Group string `json:"group"`
```

Comment:
Version also?

Reply:
would it be ok if we only use the latest version supplied?

Reply:
the operator will internally know which version of the resource it should use

Reply:
> the operator will internally know which version of the resource it should use

Yeah, all versions represent the same data.  Consumer can choose.  This schema matches https://github.com/openshift/api/blob/master/config/v1/types_cluster_operator.go#L79 to interact with `must-gather` and `oc adm inspect`.  You may want to indicate that in the type doc

Reply:
talking to @showeimer i think we may need the `Version` field, in order to expose to our user which exact resource to update from UI 

also I think if any controller want to use dynamic clientset with this reference we may also need the version as well